datadog_checks_downloader/datadog_checks/downloader/download.py

```diff
@@ -253,23 +256,32 @@ def get_latest_version(self, standard_distribution_name, wheel_distribution_name
         except UnknownTargetError:
             raise NoSuchDatadogPackage(standard_distribution_name)
 
-        pattern = "<a href='(" + wheel_distribution_name + "-(.*?)-py2\\.py3-none-any\\.whl)'>(.*?)</a><br />"
-        versions = []
+        pattern = "<a href='(" + wheel_distribution_name + "-(.*?)-(.*?)-none-any\\.whl)'>(.*?)</a><br />"
+        packages = defaultdict(list)
 
         with open(target_abspath) as simple_index:
             for line in simple_index:
                 match = re.match(pattern, line)
                 if match:
-                    href = match.group(1)
-                    version = match.group(2)
-                    text = match.group(3)
+                    href, package_version, python_versions, text = match.groups()
+
                     if href != text:
                         raise InconsistentSimpleIndex(href, text)
                     else:
-                        # https://setuptools.readthedocs.io/en/latest/pkg_resources.html#parsing-utilities
-                        versions.append(parse_version(version))
+                        packages[package_version].append((href, python_versions.split('.')))
 
-        if not len(versions):
+        if not packages:
             raise MissingVersions(standard_distribution_name)
+
+        if not desired_package_version:
+            # https://setuptools.readthedocs.io/en/latest/pkg_resources.html#parsing-utilities
+            desired_package_version = str(max(parse_version(p) for p in packages))
+        elif desired_package_version not in packages:
+            raise MissingVersions(standard_distribution_name)
+
+        desired_python_version = 'py{}'.format(sys.version_info[0])
```

Comment:
Could we maybe use `six` here, or is this robust enough?

Reply:
This is more robust, since it accounts for future versions 😄 

Also: https://github.com/benjaminp/six/blob/aa4e90bcd7b7bc13a71dfaebcb2021f4caaa8432/six.py#L35-L37